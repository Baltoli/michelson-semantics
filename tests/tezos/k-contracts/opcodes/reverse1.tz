$contract { parameter (list string);
storage (list string);
code { CAR; NIL string; SWAP;
       ITER {CONS};
       NIL operation; PAIR};
 } ;
$param { "c" ; "b" ; "a" } ;
$storage {""} ;
$ExpectedOutput { "a" ; "b" ; "c" } ;

