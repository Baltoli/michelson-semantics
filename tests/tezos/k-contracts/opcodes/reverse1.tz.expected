<generatedTop>
  <k>
    .
  </k>
  <stack>
    Pair .List ListItem ( "a" )
    ListItem ( "b" )
    ListItem ( "c" )
  </stack>
  <paramtype>
    list .AnnotationList string .AnnotationList
  </paramtype>
  <storagetype>
    list .AnnotationList string .AnnotationList
  </storagetype>
  <balance>
    #Mutez ( 0 )
  </balance>
  <amount>
    #Mutez ( 0 )
  </amount>
  <now>
    #Timestamp ( -1 )
  </now>
  <myaddr>
    #Address ( "InvalidMyAddr" )
  </myaddr>
  <knownaddrs>
    .Map
  </knownaddrs>
  <sourceaddr>
    #Address ( "InvalidSourceAddr" )
  </sourceaddr>
  <senderaddr>
    #Address ( "InvalidSenderAddr" )
  </senderaddr>
  <chainid>
    #ChainId ( 0 )
  </chainid>
</generatedTop>
